src/lib/stateMappings/simple.ts

```diff
@@ -1,32 +1,36 @@
-import { UiState, IndexUiState, StateMapping, RouteState } from '../../types';
+import { UiState, IndexUiState, StateMapping } from '../../types';
 
-function getIndexStateWithoutConfigure(uiState: IndexUiState): IndexUiState {
+function getIndexStateWithoutConfigure<TIndexUiState extends IndexUiState>(
+  uiState: TIndexUiState
+): Omit<TIndexUiState, 'configure'> {
   const { configure, ...trackedUiState } = uiState;
   return trackedUiState;
 }
 
 // technically a URL could contain any key, since users provide it,
 // which is why the input to this function is UiState, not something
 // which excludes "configure" as this function does.
-export default function simpleStateMapping(): StateMapping {
+export default function simpleStateMapping<
+  TUiState extends UiState = UiState
+>(): StateMapping<TUiState, TUiState> {
   return {
     stateToRoute(uiState) {
-      return Object.keys(uiState).reduce<RouteState>(
+      return Object.keys(uiState).reduce(
         (state, indexId) => ({
           ...state,
           [indexId]: getIndexStateWithoutConfigure(uiState[indexId]),
         }),
-        {}
+        {} as TUiState
       );
```

Comment:
Would this work to avoid casting the initial value?

```ts
Object.keys(uiState).reduce<TUiState>(
  (state, indexId) => ({
    ...state,
    [indexId]: getIndexStateWithoutConfigure(uiState[indexId]),
  }),
  {}
);
```

Reply:
no, as TUiState doesn't necessarily allows empty object (but we assume it does),

```
Argument of type '{}' is not assignable to parameter of type 'TUiState'.
  '{}' is assignable to the constraint of type 'TUiState', but 'TUiState' could be instantiated with a different subtype of constraint 'UiState'
```